scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,445 @@
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable parameter string size"
+version: "1.7"
+name: $T_MAX_METRIC_PROGRAMMABLE_PARAMETER_NAME_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum value description string size"
+version: "1.7"
+name: $T_MAX_VALUE_DESCRIPTION_CSTRING_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: handle
+desc: "Handle of metric programmable's object"
+version: "1.7"
+class: $tMetricProgrammable
+name: "$t_metric_programmable_exp_handle_t"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Metric Programmable properties queried using $tMetricProgrammableGetPropertiesExp"
+version: "1.7"
+class: $tMetricProgrammable
+name: $t_metric_programmable_exp_properties_t
+base: $t_base_properties_t
+members:
+    - type: char
+      name: "name[$T_MAX_METRIC_PROGRAMMABLE_NAME_EXP]"
+      desc: "[out] metric programmable name"
+    - type: char
+      name: "description[$T_MAX_METRIC_PROGRAMMABLE_DESCRIPTION_EXP]"
+      desc: "[out] metric programmable description"
+    - type: char
+      name: "component[$T_MAX_METRIC_PROGRAMMABLE_COMPONENT_EXP]"
+      desc: "[out] metric programmable component"
+    - type: uint32_t
+      name: "tierNumber"
+      desc: "[out] tier number"
+    - type: uint32_t
+      name: "domain"
+      desc: "[out] metric domain number."
+    - type: uint32_t
+      name: "parameterCount"
+      desc: "[out] number of parameters in the programmable"
+    - type: "$t_metric_group_sampling_type_flags_t"
+      name: "samplingType"
+      desc: |
+            [out] metric sampling type.
+            returns a combination of $t_metric_group_sampling_type_flags_t.
+    - type: uint32_t
+      name: "sourceId"
+      desc: "[out] unique metric source identifier(within platform)to identify the HW block where the metric is collected."
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Metric Programmable Parameter types"
+version: "1.7"
+class: $tMetricProgrammable
+name: $t_metric_programmable_param_type_exp_t
+etors:
+    - name: DISAGGREGATION
+      desc: "Metric is disaggregated."
+    - name: LATENCY
+      desc: "Metric for latency measurement."
+    - name: NORMALIZATION_UTILIZATION
+      desc: "Produces normalization in percent using raw_metric * 100 / cycles / instance_count."
+    - name: NORMALIZATION_AVERAGE
+      desc: "Produces normalization using raw_metric / instance_count."
+    - name: NORMALIZATION_RATE
+      desc: "Produces normalization average using raw_metric / timestamp."
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Supported value description types"
+version: "1.7"
+name: $t_value_type_exp_desc_t
```

Comment:
descriptors generally contain members, since this is enum this should be renamed

Reply:
Ok.
Please share if there is a guideline for it.
Thanks

Reply:
renamed to info